### PR TITLE
Schema getter setter tests

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
The function that sets the schema in arguments needs tests.

# Changes
- Define a schema getter - useful for checking whether the setter has worked correctly.
- Write tests for the getter and setter.
- Fix bug.
- Exercise the tests in CI

# Tests
Tests form the majority of this PR.

# Todos

- [x] Add entry to changelog (if necessary).
  - Covered by an existing entry

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -364,7 +373,27 @@ jobs:
       - name: Getting network config works
         run: scripts/network-config.test
```
Comment (bitdivine) [inline .github/workflows/checks.yml:374]: Reformatted as the list below is too long for one line.

File: scripts/nns-dapp/migration-test [hunk 2]
```diff
@@ -111,7 +111,7 @@ get_assets() {
     get_wasm "${!WASM}" "$(working_wasm "$config")"
     get_argfile "${!ARGFILE}" "$(working_arguments "$config")"
     test -z "${!SCHEMA}" || {
-      set_schema "${!SCHEMA}" <"$(working_arguments "$config")" | sponge "$(working_arguments "$config")"
+      set_arguments_schema "${!SCHEMA}" <"$(working_arguments "$config")" | sponge "$(working_arguments "$config")"
```
Comment (bitdivine) [inline scripts/nns-dapp/migration-test:114]: Renamed for clarity.

File: scripts/nns-dapp/migration-test.getters [hunk 3]
```diff
@@ -69,15 +69,21 @@ get_argfile() {
...
 # Reads from stdin and writes to stdout.
 # TODO: Unit Test
-set_schema() {
+set_arguments_schema() {
   local schema="${1}" # TODO: Add linter for schema.
   # TODO: Finalize support for compound types in yaml2candid.  Then we can use something like this:
   # idl2json | schema="$schema" jq '.[0].schema = (env.schema)' | yaml2candid --did rs/backend/nns-dapp.did  --typ 'opt Config'
   # This is simple but can break if the format is not as expected:
-  awk -v schema="$schema" '/^ *schema *= *opt *variant/{ printf "  schema = opt variant { %s };\n", schema; next }{print}' out/nns-dapp-arg-local.did
+  awk -v schema="$schema" '/^ *schema *= *opt *variant/{ printf "  schema = opt variant { %s };\n", schema; next }{print}'
```
Comment (bitdivine) [inline scripts/nns-dapp/migration-test.getters:86]: The arguments are meant to be read from stdin, not from the file I was testing with when writing the code. :facepalm:

**Review by lmuntaner — APPROVED** [review]
LGTM, rubber stump!

**Pull request merged**